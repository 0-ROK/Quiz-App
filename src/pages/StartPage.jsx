import React from "react";
import { useNavigate } from "react-router";
import Button from "@mui/material/Button";
import { useResetRecoilState, useSetRecoilState } from "recoil";
import { pageState } from "../states/recoilPageState";
import { useEffect } from "react";
import { timerState } from "../states/recoilTimerState";
import { resolvedQuizState } from "../states/recoilResolvedQuizState";

const StartPage = () => {
  const navigate = useNavigate();
  const setPage = useSetRecoilState(pageState);
  const resetPage = useResetRecoilState(pageState);
  const resetTime = useResetRecoilState(timerState);
  const resetResolvedQuiz = useResetRecoilState(resolvedQuizState);

  useEffect(() => {
    resetPage();
    resetTime();
    resetResolvedQuiz();
  }, []);

  return (
    <div className="App">
      <div className="App-header">
        <p>즐거운 퀴즈!!</p>
        <Button
          variant="contained"
          onClick={() => {
            setPage((page) => page + 1, navigate("/quiz"));
          }}
        >
          퀴즈 시작해볼까요?
        </Button>
      </div>
    </div>
  );
};

export default StartPage;
