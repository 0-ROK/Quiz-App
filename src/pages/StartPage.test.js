import { render, screen, fireEvent } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import { BrowserRouter as Router } from "react-router-dom";
import StartPage from "./StartPage";

describe("<StartPage/>", () => {
  it("is start page snapshot", () => {
    const utils = render(
      <RecoilRoot>
        <Router>
          <StartPage />
        </Router>
      </RecoilRoot>
    );
    expect(utils.container).toMatchSnapshot();
  });

  it("시작 버튼을 누르면 /quiz 페이지로 이동한다.", () => {
    render(
      <RecoilRoot>
        <Router>
          <StartPage />
        </Router>
      </RecoilRoot>
    );

    const startButton = screen.getByRole("button", {
      name: "퀴즈 시작해볼까요?",
    });

    fireEvent.click(startButton);

    expect(global.window.location.pathname).toBe("/quiz");
  });
});
