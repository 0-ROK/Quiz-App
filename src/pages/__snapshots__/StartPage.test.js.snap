// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartPage/> is start page snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="App-header"
    >
      <p>
        즐거운 퀴즈!!
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        퀴즈 시작해볼까요?
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
